### ISSUE TITLE
Add post object as the third argument of `wpp_fix_post_date` function

### ISSUE THREAD

**Issue by had1z** [wordpress-parsi/wp-parsidate#121]
Wordpress `get_the_date()` function has a third argument which is the `WP_Post` object; if the argument is not passed to the function, then it will use the global `$post` variable. 
At the end of the function when it returns the filtered result, it passes the variable through the `get_the_date` filter which has also a third parameter:
```
/**
 * Retrieve the date on which the post was written.
 *
 * Unlike the_date() this function will always return the date.
 * Modify output with the {@see 'get_the_date'} filter.
 *
 * @since 3.0.0
 *
 * @param string      $format Optional. PHP date format. Defaults to the 'date_format' option.
 * @param int|WP_Post $post   Optional. Post ID or WP_Post object. Default current post.
 * @return string|false Date the current post was written. False on failure.
 */
function get_the_date( $format = '', $post = null ) {
	$post = get_post( $post );

	if ( ! $post ) {
		return false;
	}

	$_format = ! empty( $format ) ? $format : get_option( 'date_format' );

	$the_date = get_post_time( $_format, false, $post, true );

	/**
	 * Filters the date a post was published.
	 *
	 * @since 3.0.0
	 *
	 * @param string      $the_date The formatted date.
	 * @param string      $format   PHP date format.
	 * @param int|WP_Post $post     The post object or ID.
	 */
	return apply_filters( 'get_the_date', $the_date, $format, $post );
}
```

So in the applying function `wpp_fix_post_date` we need to accept a third argument, because the Wordpress `get_the_date()` function's use case is not only in the post loops, some would call it outside the loop.

**Comment by man4toman** [comment]
Hi @had1z 
Thanks for reporting this issue.
We will fix this asap.

Regards

**Comment by had1z** [comment]
Hi @man4toman, thank you for your response.
I can add this in a PR.

**Comment by man4toman** [comment]
It would be very nice if can do it.

**Comment by hamidrezayazdani** [comment]
Resolved in #125
